### DISCUSSION TITLE
IDA flare-emu 教程 - B1TG

### DISCUSSION THREAD

**Discussion by giscus** [b1tg/b1tg.github.io#1] [Discuss]
# IDA flare-emu 教程 - B1TG

0x00 前言 在逆向工程中，常常会遇到加/解密函数，运气好的话发现是常见算法，找个在线工具直接解就好；运气不好的话就得慢慢厘清算法细节，并尝试写脚本

https://b1tg.github.io/post/flare-emu-tutorial/

**Comment by b1tg** [comment]
有评论功能啦 🍟

**Comment by mark0smith** [comment]
您好，请问您方便提供下从fake_virus_decrypt_in_mem.cpp编译的参数设置吗？或者是编译后的`demo.exe`文件？

我编译之后，修改示例A脚本`xor_str`函数地址运行得到的`rax`寄存器值全是`\x00`
```
00000000: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
00000010: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
```

**Reply by b1tg** [reply]
我当时用 Visual Studio 编译的，应该没改过参数。你可以试试调用 emulateRange 函数时指定 instructionHook 参数来做一下调试和定位。

**Reply by mark0smith** [reply]
好的，我在分析其他程序时使用`flare-emu`成功模拟运行了程序内的函数，感谢您的文章和解答。
